x-pack/plugins/apm/public/components/app/service_inventory/index.tsx

```diff
@@ -39,19 +39,24 @@ function useServicesFetcher() {
   const { urlParams, uiFilters } = useUrlParams();
   const { core } = useApmPluginContext();
   const upgradeAssistantHref = useUpgradeAssistantHref();
-  const { start, end } = urlParams;
+  const { environment, start, end } = urlParams;
   const { data = initialData, status } = useFetcher(
     (callApmApi) => {
       if (start && end) {
         return callApmApi({
           endpoint: 'GET /api/apm/services',
           params: {
-            query: { start, end, uiFilters: JSON.stringify(uiFilters) },
+            query: {
+              environment,
+              start,
+              end,
+              uiFilters: JSON.stringify(uiFilters),
```

Comment:
What's left in `uiFilters` now that environment is extracted? Hint: can we remove it? 😋

Reply:
... ah, I guess it's the `kuery`? Would love it if that can be extracted from uiFilters. Tricky because of RUM dependency?

Reply:
`kuery` is the only thing left (outside of RUM) but I'm removing `uiFilters` and moving `kuery` in a separate PR, since this one is already huge.

Reply:
Makes sense